### ISSUE TITLE
Encrypted Assertion PrincipalName

### ISSUE THREAD

**Issue by moakley7842** [akunzai/GSS.Authentication.CAS#17]
What in ticket validation determines whether the assertion PrincipalName is encrypted or not.  Different behavior on different machines where on one the PrincipalName is clear while on other machines the PrincipalName is encrypted.

**Comment by akunzai** [comment]
Please provide the code to reproduce your problem

**Comment by moakley7842** [comment]
We were able to dig a little further.  Using the log of our CAS Server we believe that there may be a server setup problem, not with your middleware.  For the last 3 actions that get logged when we fail

- SERVICE_TICKET_CREATED the who and what look fine.
- SERVICE_ACCESS_ENFORCEMENT_TRIGGERED
      - Who is set to audit:unknown
      - What includes principal=SimplePrincipal(id=q2A1e ...  What could be a transient Id
- SERVICE_TICKET_VALIDATED the who is set to the value of what was in the simple principal id.

On the one machine where we succeed the Id is what we expect and so is the who on the last action.

So no I do not believe that this is a problem with your middleware, but a CAS server configuration problem.
